torch/nn/modules/module.py

```diff
@@ -1426,6 +1429,34 @@ def _register_load_state_dict_pre_hook(self, hook, with_module=False):
         self._load_state_dict_pre_hooks[handle.id] = hook
         return handle
 
+    def register_load_state_dict_post_hook(self, hook):
+        r"""Registers a post hook to be run after module's ``load_state_dict``
+        is called. The hook will be called everytime after :func:`load_state_dict`
+        has loaded a ``state_dict`` into the ``self`` module.
+        It should have the following signature::
+            hook(module, incompatible_keys) -> None
+        The ``module`` argument is the current module that this hook is registered
+        on, and the ``incompatible_keys`` argument is a ``NamedTuple`` consisting
+        of attributes ``missing_keys`` and ``unexpected_keys``. ``missing_keys``
+        is a ``list`` of ``str`` containing the missing keys and
+        ``unexpected_keys`` is a ``list`` of ``str`` containing the unexpected keys.
+        If the hook wishes to modify the ``NamedTuple`` ``missing_keys``
+        or ``unexpected_keys`` returned by ``load_state_dict``, the hook
+        can simply modify the ``NamedTuple`` argument it is invoked with.
+        Note that if the hook modifies (adds) entries to ``missing_keys`` or
```

Comment:
@jbschlosser Added blurb about how it works with strict=True raising errors.

Reply:
It's also possible that the hook removes keys from `missing_keys` or `unexpected_keys`, avoiding what otherwise would be an error with `strict=True`, is that right?

If so, maybe the phrasing can be a bit more general e.g.
``` Note that the checks performed when calling :func:`load_state_dict` with ``strict=True`` are affected by modifications the hook makes to ``missing_keys`` or ``unexpected_keys``, as expected. Additions to either set of keys will result in an error being thrown when ``strict=True``, and clearing out both missing and unexpected keys will avoid an error, allowing the behavior of the strict flag to be customized as needed by the hook.```

Or maybe this is obvious idk :p

Reply:
oh yeah, will add the test of this too